homeassistant/components/camera/verisure.py

```diff
@@ -0,0 +1,101 @@
+"""
+Camera that loads a picture from a local file.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/camera.verisure/
+"""
+import logging
+import os
+
+from homeassistant.components.camera import Camera
+from homeassistant.const import EVENT_HOMEASSISTANT_STOP
+from homeassistant.components.verisure import HUB as hub
+from homeassistant.components.verisure import CONF_SMARTCAM
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Camera."""
+    if not int(hub.config.get(CONF_SMARTCAM, 1)):
+        return False
+    directory_path = hass.config.config_dir
+    if not os.access(directory_path, os.R_OK):
+        _LOGGER.error("file path %s is not readable", directory_path)
+        return False
+    hub.update_smartcam()
+    smartcams = []
+    smartcams.extend([
+        VerisureSmartcam(value.deviceLabel, directory_path)
+        for value in hub.smartcam_status.values()])
+    add_devices(smartcams)
+    hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP,
+                         VerisureSmartcam.delete_image)
+
+
+class VerisureSmartcam(Camera):
+    """Local camera."""
+
+    def __init__(self, device_id, directory_path):
+        """Initialize Verisure File Camera component."""
+        super().__init__()
+
+        self._device_id = device_id
+        self._directory_path = directory_path
+        self._image = None
+        self._image_id = None
+        self._smartcam_dict = hub.update_smartcam_imagelist()
+
+    def camera_image(self):
+        """Return image response."""
+        self.check_imagelist()
+        _LOGGER.debug('trying to open %s', self._image)
+        with open(self._image, 'rb') as file:
+            return file.read()
+
+    def check_imagelist(self):
+        """Check the contents of the image list."""
+        self._smartcam_dict = hub.update_smartcam_imagelist()
```

Comment:
this is not really what I meant. you have to check if the device has a key in the dict:
```
hub.update_smartcam_imagelist()
if self._device_id not in hub.update_smartcam_imagelist or not hub.sartcam_imagelist[self._device_id]:
    return
images = hub.smartcam_dict[self._device_id]
```
The first check in the ```if``` controls that the device has a key in the dictionay and hte seckond check is that it has at least one item in the list

You can remove ```self._smartcam_dict``` and use ```hub.smartcam_dict``` insead. No need to store in to two places.